resources/api/resource.py

```diff
@@ -245,19 +254,42 @@ def filter(self, qs, value):
 class ParentCharFilter(ParentFilter):
     field_class = forms.CharField
 
+
 class DRFFilterBooleanWidget(django_filters.widgets.BooleanWidget):
     def render(self, *args, **kwargs):
         return None
 
+
 class ResourceFilterSet(django_filters.FilterSet):
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
+        super().__init__(*args, **kwargs)
+
     purpose = ParentCharFilter(name="purposes__id", lookup_type='iexact')
     type = django_filters.CharFilter(name="type__id", lookup_type='iexact')
     people = django_filters.NumberFilter(name="people_capacity", lookup_type='gte')
     need_manual_confirmation = django_filters.BooleanFilter(name="need_manual_confirmation", widget=DRFFilterBooleanWidget)
+    is_favorite = django_filters.MethodFilter(widget=django_filters.widgets.BooleanWidget())
 
     class Meta:
         model = Resource
-        fields = ['purpose', 'type', 'people', 'need_manual_confirmation']
+        fields = ['purpose', 'type', 'people', 'need_manual_confirmation', 'is_favorite']
+
+    def filter_is_favorite(self, queryset, value):
+        favorites = self.user.favorite_resources.values_list('id') if self.user.is_authenticated() else []
```

Comment:
I think you could put the DB to work by using `queryset.filter(favorited_by=self.user)` instead of fetching the favorite ids separately.